openml-lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMModelCreator.java

```diff
@@ -0,0 +1,293 @@
+                                                     final Map<String, String> params) {
+        final ImmutableList.Builder<ParamValidationError> errorsBuilder = ImmutableList.builder();
+
+        errorsBuilder.addAll(validateModelPathToTrain(pathToPersist));
+        errorsBuilder.addAll(checkParams(
+                LightGBMAlgorithms.LIGHTGBM_BINARY_CLASSIFIER.getAlgorithmDescriptor(), params));
+        validateCategoricalSchema(schema).ifPresent(errorsBuilder::add);
+
+        if (schemaHasStringFields(schema)) {
+            errorsBuilder.add(new ParamValidationError(ERROR_MSG_SCHEMA_HAS_STRING_FIELDS));
+        }
+
+        if (getNumTargetClasses(schema).orElse(-1) != 2) {
+            errorsBuilder.add(new ParamValidationError(ERROR_MSG_NON_BINARY_TARGET));
+        }
+
+        if (params.get(BOOSTING_TYPE_PARAMETER_NAME).equals("rf") && baggingDisabled(params)) {
+            logger.warn("RF requires bagging. Set bagging fraction < 1 and bagging frequency > 0.");
+            errorsBuilder.add(new ParamValidationError(ERROR_MSG_RANDOM_FOREST_REQUIRES_BAGGING));
+        }
+
+        return errorsBuilder.build();
+    }
+
+    /**
+     * Checks if bagging is disabled.
+     *
+     * @param params LightGBM parameters
+     * @return true if disabled, false otherwise.
+     */
+    private boolean baggingDisabled(final Map<String, String> params) {
+
+        final double epsilon = 1e-60;
+
+        final double freq = Double.parseDouble(params.get(BAGGING_FREQUENCY_PARAMETER_NAME));
+        final double fraction = Double.parseDouble(params.get(BAGGING_FRACTION_PARAMETER_NAME));
+        return ((Math.abs(freq - 0) < epsilon) || (Math.abs(1 - fraction) < epsilon));
+    }
+
+    @Override
+    public LightGBMBinaryClassificationModel loadModel(final Path modelPath,
+                                                       final DatasetSchema schema)  throws ModelLoadingException {
+
+        // If modelPath is a directory, look for the model file inside:
+        Path modelFilePath = modelPath;
+        if (Files.isDirectory(modelPath)) {
+            modelFilePath = modelPath.resolve(MODEL_BINARY_RESOURCE_FILE_NAME);
+        }
+
+        logger.info("Loading LightGBM model from " + modelFilePath.toAbsolutePath().toString());
+        final LightGBMBinaryClassificationModel model = new LightGBMBinaryClassificationModel(
+                modelFilePath, schema
+        );
+
+        // LightGBM considers binary classification as a special case of 1 class:
+        if (! model.isModelBinary()) {
+            throw new ModelLoadingException(ERROR_MSG_CANNOT_LOAD_NON_BINARY_LIGHTGBM_MODEL);
+        }
+
+        if (model.getBoosterNumFeatures() != schema.getPredictiveFields().size()) {
+            throw new ModelLoadingException(ERROR_MSG_SCHEMA_WITH_WRONG_PREDICTIVE_FIELDS_SIZE);
+        }
+
+        /*
+          TODO: https://issues.feedzai.com/browse/PULSEDEV-30615 CHECK FEATURE NAMES ORDER IN MODEL AGAINST LOADED SCHEMA.
```

Comment:
You can remove these TODOs. We already have them internally anyway.

Reply:
You mean in JIRA? We're not going to keep with 2 codebases right @JPDSousa ?

Reply:
this provider was removed from the internal repository

Reply:
Not sure, but  I think we have the task on Jira, we can track based on there. Therefore I removed this TODO